prime-router/src/main/kotlin/cli/LivdTableDownload.kt

```diff
@@ -123,4 +137,96 @@ class LivdTableDownload() : CliktCommand(
     }
+
+    /**
+     * Extracts LIVD table from sheet [sheetName] of the input Excel formate [inputfile] file, converts to
+     * csv format, and output to the CSV format [outpufile] file.
+     * @param sheetName is the LOINC Mapping sheet from the downloaded LOINC data code file.
+     * @param inputfile is the input Excel file name.
+     * @param inputfile is the output CSV file name.
+     * @return true for success and fales for failure.
+     */
+    private fun extractsLivdTable(sheetName: String, inputfile: String, outputfile: String): Boolean {
+
+        try {
+            val data = StringBuffer() // Buffer and output file for CSV data
+            val fileInputStream = FileInputStream(File(inputfile))
+            var workbook: Workbook? = null
+            val ext: String = FilenameUtils.getExtension(inputfile)
+            if (ext.equals("xlsx", ignoreCase = true)) {
+                workbook = XSSFWorkbook(fileInputStream)
+            } else if (ext.equals("xls", ignoreCase = true)) {
+                workbook = HSSFWorkbook(fileInputStream)
+            }
+
+            // Check for output file duplication
+            if (File(outputfile).exists()) {
+                val c = TermUi.prompt("\t$outputfile file is already existed: You want to overwrite it (y/n)?", "n")
+                if (c?.lowercase() == "n") {
+                    return false
+                } else {
+                    TermUi.echo("\tThe $outputfile file is overwriting.")
+                }
+            }
+
+            val fileOutputStream = FileOutputStream(File(outputfile))
+
+            // Get the LOINC Mapping sheet
+            val sheet: Sheet? = workbook!!.getSheet(sheetName)
+            if (sheet == null) {
+                print("\tERROR: Sheet \"$sheetName\" doesn't exist in the $inputfile file.")
+                return false
+            }
+
+            val rowStart = sheet.getFirstRowNum()
+            val rowEnd = sheet.getLastRowNum()
+
+            for (rowNum in rowStart until rowEnd + 1) {
+                val row: Row = sheet.getRow(rowNum) ?: continue
+
+                val lastColumn: Short = row.lastCellNum
+                for (cn in 0 until lastColumn) {
+                    var delimiterChar = ","
+                    if (cn + 1 == lastColumn.toInt()) delimiterChar = ""
+
+                    val cell = row.getCell(cn, Row.MissingCellPolicy.RETURN_BLANK_AS_NULL)
+
+                    if (cell == null) {
+                        data.append("" + delimiterChar)
+                    } else {
+                        when (cell.cellType) {
+                            CellType.BOOLEAN -> data.append(cell.booleanCellValue.toString() + delimiterChar)
+                            CellType.NUMERIC -> data.append(cell.numericCellValue.toString() + delimiterChar)
+                            CellType.STRING -> {
+                                val stringValue = cell.stringCellValue.replace("*", "")
```

Comment:
Do not replace all *.  You can only remove it if it is the last character.  Also wondering if we should check if you are on the specific column that could have the *, but unsure if that makes sense.

Reply:
Done.